### PR TITLE
#M21. Add functionality to set maximum number of registrations.

### PR DESCRIPTION
**Pull request by stefdworschak**
## Description
- Adding functionality to allow limiting the number of registrations
- Sprucing up the external user login page

## Pull request type

- [x] #M01 (Miscellaneous User Story #01)
- [ ] #P02 (Participant User Story #02)
- [ ] #S03 (Staff User Story #03)
- [ ] #A04 (Admin User Story #04)
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue
#197

## Configuration instructions
- Run migrations

### CODE REVIEW

**Review by TravelTimN — APPROVED** [review]
A few areas for minor improvements, and some suggested recommendations (optional). Apart from that, LGTM and approved with those small adjustments where needed.

File: hackathon/templates/hackathon/includes/enroll_card.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{% if not hackathon.max_participants_reached %}
+<form id="enroll-form" action="{% url 'hackathon:enroll_toggle' %}" method="POST">
+    {% csrf_token %}
+    <!--It only sends the hackathon ID.-->
+    <input type="hidden" name="hackathon-id" value="{{ hackathon.id }}">
+    <!-- Must be one long string for the JS script to switch words correctly.
+    An alternative would be to have two different buttons.-->
```
[the remark this answers is not in the document]
Comment (stefdworschak) [reply hackathon/templates/hackathon/includes/enroll_card.html:7]: Sorry, I did not leave that comment. Just copied it from the other enrollment form, but do I need to do anything here?

File: hackathon/forms.py [hunk 2]
```diff
@@ -88,12 +88,17 @@ class HackathonForm(forms.ModelForm):
         })
     )
     is_public = forms.BooleanField(required=False)
+    max_participants = forms.IntegerField(
+        label="Max. Number Of Participants (leave empty for no max)",
```
[the remark this answers is not in the document]
Comment (stefdworschak) [reply hackathon/forms.py:92]: Updated as suggested.

File: hackathon/templates/hackathon/hackathon_view.html [hunk 3]
```diff
@@ -98,8 +98,13 @@
                                 {% if request.user.user_type|is_types:authorised_types %}
                                 <p class="hackathon-sub-details"><i class="far fa-question-circle"></i> Status: {{ hackathon.status }}</p>
                                 <p class="hackathon-sub-details"><i class="fas fa-chalkboard-teacher"></i> Organiser: {{ hackathon.organiser }}</p>
+                                <p class="hackathon-sub-details"><i class="fas fa-users"></i>
+                                    Max. Participants: {% if hackathon.max_participants %}{{ hackathon.max_participants }}{% else %}Unlimited{% endif %}
```
[the remark this answers is not in the document]
Comment (stefdworschak) [reply hackathon/templates/hackathon/hackathon_view.html:102]: Updated as suggested.

File: hackathon/templates/hackathon/hackathon_view.html [hunk 4]
```diff
@@ -98,8 +98,13 @@
...
+                                    {% if hackathon.max_participants_reached %}(Max. Reached){% endif %}
```
[the remark this answers is not in the document]
Comment (stefdworschak) [reply hackathon/templates/hackathon/hackathon_view.html:103]: Updated as suggested.

File: templates/registration/login.html [hunk 5]
```diff
@@ -1,9 +1,18 @@
 {% extends 'base.html' %}
 {% block content %}
-<h2>Admin Login</h2>
+<h2>External Login</h2>
 <form action="/accounts/admin/" method="POST">
     {% csrf_token %}
     {{form.as_p}}
-    <input type="submit" class="btn btn-primary">
+    <input type="submit" class="btn btn-primary btn-sm" value="Sign In">
+    <a href="http://localhost:8000/accounts/password/reset/" class="ml-3">Forgot Your Password?</a>
```
[the remark this answers is not in the document]
Comment (stefdworschak) [reply templates/registration/login.html:8]: Fixed link.

File: templates/allauth/socialaccount/snippets/slack_signin.html [hunk 6]
```diff
@@ -15,5 +15,9 @@
         <a href="http://bit.ly/Slack_Code_Conduct_CI"
             target="_blank">Community Guidelines</a>.
     </p>
+    <p class="external-signin">
+        If you are not part of th Code Institute Slack Workspace, but have an
```
[the remark this answers is not in the document]
Comment (stefdworschak) [reply templates/allauth/socialaccount/snippets/slack_signin.html]: Fixed the spelling mistake.

File: hackathon/templates/hackathon/includes/enroll_card.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (TravelTimN) [reply hackathon/templates/hackathon/includes/enroll_card.html:7]: Right - wasn't aware of that just being a copy/paste job.
No, I don't think so... we could probably just remove the comment in that case, doesn't seem to be necessary here I believe.

[the remark this answers is not in the document]
Comment (stefdworschak) [reply hackathon/templates/hackathon/includes/enroll_card.html]: Removed unneeded comments

**Pull request merged**